abi-versions/v0.2.1/README.md

```diff
@@ -0,0 +1,2166 @@
+
+Returned `status` value is:
+- `OK` on success.
+- `BAD_ARGUMENT` for unknown `map_id`.
+- `INVALID_MEMORY_ACCESS` when `serialized_pairs_data` and/or
+  `serialized_pairs_size` point to invalid memory address.
+
+
+#### `proxy_get_header_map_value`
+
+* params:
+  - `i32 (`[`proxy_map_type_t`]`) map_id`
+  - `i32 (const char *) key_data`
+  - `i32 (size_t) key_size`
+  - `i32 (uint8_t **) return_value_data`
+  - `i32 (size_t *) return_value_size`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Retrieves value (`return_value_data`, `return_value_size`) of the key
+(`key_data`, `key_value`) from the map `map_id`.
+
+Returned `status` value is:
+- `OK` on success.
+- `BAD_ARGUMENT` for unknown `map_id`.
+- `NOT_FOUND` when the requested key was not found.
+- `INVALID_MEMORY_ACCESS` when `key_data`, `key_size`,
+  `return_value_data` and/or `return_value_size` point to
+  invalid memory address.
+
+
+#### `proxy_add_header_map_value`
+
+* params:
+  - `i32 (`[`proxy_map_type_t`]`) map_id`
+  - `i32 (const char *) key_data`
+  - `i32 (size_t) key_size`
+  - `i32 (const uint8_t *) value_data`
+  - `i32 (size_t) value_size`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Adds key (`key_data`, `key_size`) with value (`value_data`,
+`value_size`) to the map `map_id`.
+
+Returned `status` value is:
+- `OK` on success.
+- `BAD_ARGUMENT` for unknown `map_id`.
+- `INVALID_MEMORY_ACCESS` when `key_data`, `key_size`, `value_data`
+  and/or `value_size` point to invalid memory address.
+
+
+#### `proxy_replace_header_map_value`
+
+* params:
+  - `i32 (`[`proxy_map_type_t`]`) map_id`
+  - `i32 (const char *) key_data`
+  - `i32 (size_t) key_size`
+  - `i32 (const uint8_t *) value_data`
+  - `i32 (size_t) value_size`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Adds or replaces key's (`key_data`, `key_value`) value with the provided
+value (`value_data`, `value_size`) in the map `map_id`.
+
+Returned `status` value is:
+- `OK` on success.
+- `BAD_ARGUMENT` for unknown `map_id`.
+- `INVALID_MEMORY_ACCESS` when `key_data`, `key_size`, `value_data`
+  and/or `value_size` point to invalid memory address.
+
+
+#### `proxy_remove_header_map_value`
+
+* params:
+  - `i32 (`[`proxy_map_type_t`]`) map_id`
+  - `i32 (const char *) key_data`
+  - `i32 (size_t) key_size`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Removes the key (`key_data`, `key_value`) from the map `map_id`.
+
+Returned `status` value is:
+- `OK` on success (including the case when the requested key didn't exist).
+- `BAD_ARGUMENT` for unknown `map_id`.
+- `INVALID_MEMORY_ACCESS` when `key_data` and/or `key_size` point to
+  invalid memory address.
+
+
+## Common HTTP and TCP stream operations
+
+### Functions exposed by the host
+
+#### `proxy_continue_stream`
+
+* params:
+  - `i32 (`[`proxy_stream_type_t`]`) stream_type`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Resumes processing of paused `stream_type`.
```

Comment:
This has to be called if processing was stopped with StopAndBuffer.

Reply:
This is already documented in the callbacks that can stop processing.